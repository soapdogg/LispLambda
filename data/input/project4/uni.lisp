(DEFUN UNI (S1 S2)
    (COND   ((NULL S1) S2)
            ((NULL S2) S1)
            (T (COND
                ((MEM (CAR S1) S2)
                    (UNI (CDR S1) S2))
                (T
                    (CONS (CAR S1) (UNI (CDR S1) S2))
                )
            ))
    )
)

(DEFUN MEM (X LIST)
    (COND   ((NULL LIST) NIL)
            (T (COND    ((EQ X (CAR LIST)) T)
                        (T (MEM X (CDR LIST)))
               )
            )
    )
)

(UNI () ())
(UNI (QUOTE (1)) ())
(UNI () (QUOTE (1 2)))
(UNI (QUOTE (1 2 3)) ())
(UNI () (QUOTE (4 5 6)))
(UNI (QUOTE (1 56 34 23 459 3934 454854 9239 34934 845498 3434 9595)) (QUOTE (1 3 4 45 23 9595 67 69 0 343)))