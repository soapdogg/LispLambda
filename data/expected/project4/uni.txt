NIL
(1)
(1 2)
(1 2 3)
(4 5 6)
(56 34 459 3934 454854 9239 34934 845498 3434 1 3 4 45 23 9595 67 69 0 343)